gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -220,6 +219,10 @@ void finalizeJobStateBeforeCommit() {
   void commit() throws IOException {
     this.datasetStatesByUrns = Optional.of(this.jobState.createDatasetStatesByUrns());
     boolean allDatasetsCommit = true;
+    boolean shouldCommitDataInJob = shouldCommitDataInJob();
+    if (!shouldCommitDataInJob) {
```

Comment:
That `if` block is inside the `for` loop, so this log message will be printed multiple times if there are multiple datasets.